pkg/webhook/webhook.go

```diff
@@ -832,3 +892,43 @@ func SetInjectHugepageDownApi(hugepageFlag bool) {
 func SetHonorExistingResources(resourcesHonorFlag bool) {
 	honorExistingResources = resourcesHonorFlag
 }
+
+// SetCustomizedInjections sets additional injections to be applied in Pod spec
+func SetCustomizedInjections(injections *corev1.ConfigMap) {
+	var patch jsonPatchOperation
+	var cusPatchs = cusInjects.Patchs
+
+	for k, v := range injections.Data {
+		existValue, exists := cusPatchs[k]
+		// unmarshall customized injection to json patch
+		err := json.Unmarshal([]byte(v), &patch)
+		if err != nil {
+			glog.Errorf("Failed to unmarshall customized injection: %v", v)
+			continue
+		}
+		// metadata.Annotations is the only supported field for customization
+		// jsonPatchOperation.Path should be "/metadata/annotations" when customizing pod annotation
+		if patch.Path != "/metadata/annotations" {
+			glog.Errorf("Path: %v is not supported, only /metadata/annotations can be customized", patch.Path)
+			continue
+		}
+		if !exists || !reflect.DeepEqual(existValue, patch) {
+			glog.Infof("Initializing customized injections with key: %v, value: %v", k, v)
+			cusPatchs[k] = patch
+		}
+	}
+	// remove stale entries from customized configMap
+	for k, _ := range cusPatchs {
+		if _, ok := injections.Data[k]; ok {
+			continue
+		}
+		glog.Infof("Removing stale entry: %v from customized injections", k)
+		delete(cusPatchs, k)
+	}
+
+	// lock for writing
+	cusInjects.Lock()
+	defer cusInjects.Unlock()
+	cusInjects.Patchs = cusPatchs
```

Comment:
no need to do this assignment.

https://play.golang.org/p/pHs2-7u6VXW

Reply:
Done